### PR TITLE
Add ARM template deploying Web App with diagnostics logging to Blob Container

### PR DESCRIPTION
**Pull request by macborowy**
This template deploys a Web App with diagnostics logging to Storage Account Blob Container.

### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [X] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

* Add ARM template deploying Web App with diagnostics logging to Blob Container

### CODE REVIEW

**Comment by macborowy** [comment]
Is something wrong with this PR? Should I improve it?

File: 301-web-app-diagnostics-logs-blob-container/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+            "kind": "[parameters('storageAccountKind')]",
+            "resources": [
+                {
+                    "name": "[concat('default/', variables('blobContainerName'))]",
+                    "type": "blobServices/containers",
+                    "apiVersion": "2018-02-01",
+                    "properties": {
+                        "publicAccess": "Blob"
+                    },
+                    "dependsOn": [
```
Comment (bmoore-msft) [inline 301-web-app-diagnostics-logs-blob-container/azuredeploy.json]: move this above properties

File: 301-web-app-diagnostics-logs-blob-container/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+                        "[concat('Microsoft.Storage/storageAccounts/', parameters('storageAccountName'))]"
+                    ]
+                }
+            ]
+        },
+        {
+            "apiVersion": "2016-09-01",
```
Comment (bmoore-msft) [inline 301-web-app-diagnostics-logs-blob-container/azuredeploy.json]: update apiVersion

File: 301-web-app-diagnostics-logs-blob-container/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+            "type": "Microsoft.Web/serverfarms",
+            "name": "[parameters('appServicePlanName')]",
+            "location": "[parameters('location')]",
+            "sku": {
+                "Name": "[parameters('appServicePlanSkuName')]"
+            }
+        },
+        {
+            "apiVersion": "2016-08-01",
```
Comment (bmoore-msft) [inline 301-web-app-diagnostics-logs-blob-container/azuredeploy.json]: update apiVersion

File: 301-web-app-diagnostics-logs-blob-container/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+                "[concat('Microsoft.Web/serverfarms/', parameters('appServicePlanName'))]",
+                "[concat('Microsoft.Storage/storageAccounts/', parameters('storageAccountName'))]"
+            ],
+            "properties": {
+                "name": "[parameters('webAppName')]",
+                "serverFarmId": "[concat('/subscriptions/', subscription().id,'/resourcegroups/', resourceGroup().name, '/providers/Microsoft.Web/serverfarms/', parameters('appServicePlanName'))]"
+            },
+            "resources": [
+                {
+                    "apiVersion": "2015-08-01",
```
Comment (bmoore-msft) [inline 301-web-app-diagnostics-logs-blob-container/azuredeploy.json]: update apiVersion

File: 301-web-app-diagnostics-logs-blob-container/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+                    "type": "config",
+                    "name": "logs",
+                    "dependsOn": [
+                        "[concat('Microsoft.Web/sites/', parameters('webAppName'))]"
+                    ],
+                    "properties": {
+                        "applicationLogs": {
+                            "azureBlobStorage": {
+                                "level": "[parameters('diagnosticsLogsLevel')]",
+                                "sasUrl": "[concat('https://', parameters('storageAccountName'), '.blob.core.windows.net/', variables('blobContainerName'), '?', listAccountSas(parameters('storageAccountName'), '2018-02-01', variables('listAccountSasRequestContent')).accountSasToken)]",
```
Comment (bmoore-msft) [inline 301-web-app-diagnostics-logs-blob-container/azuredeploy.json]: use reference to get the uri

**Pull request merged**